internal/datastore/postgres/locks.go

```diff
@@ -0,0 +1,32 @@
+package postgres
+
+import (
+	"context"
+)
+
+type lockID uint32
+
+const (
+	// gcRunLock is the lock ID for the garbage collection run.
+	gcRunLock lockID = 1
+)
+
+func (pgd *pgDatastore) tryAcquireLock(ctx context.Context, lockID lockID) (bool, error) {
+	// Acquire the lock.
```

Comment:
Please link to: https://www.postgresql.org/docs/current/explicit-locking.html#ADVISORY-LOCKS

Reply:
Please also mention that  pg locks are reentrant, but with a caveat

```
A lock can be acquired multiple times by its owning process; for each completed lock request there must be a corresponding unlock request before the lock is actually released
...
If a session already holds a given advisory lock, additional requests by it will always succeed, even if other sessions are awaiting the lock; this statement is true regardless of whether the existing lock hold and new request are at session level or transaction level.
```